lib/Scvmm/miq_scvmm_vm_ssa_info.rb

```diff
@@ -0,0 +1,67 @@
+
+require 'miq_winrm'
+require 'miq_scvmm_parse_powershell'
+
+class MiqScvmmVmSSAInfo
+  attr_reader :vhd, :hostname, :checkpoints, :connection, :vhd_type
+  def initialize(provider, user, pass, port = nil)
+    @checkpoints = []
+    @vhd         = nil
+    @hostname    = nil
+    @vhd_type    = nil
+    @winrm       = MiqWinRM.new
+    winrmport    = port.nil? ? 5985 : port
+    options      = {:port     => winrmport,
+                    :user     => user,
+                    :pass     => pass,
+                    :hostname => provider
+                   }
+
+    @connection  = @winrm.connect(options)
+  end
+
+  def vm_host(vm_name)
+    host_script = <<-host_eol
+    Get-SCVirtualMachine -VMMServer localhost -Name "#{vm_name}" | \
+      Select-Object -ExpandProperty Hostname
+    host_eol
+
+    # TODO: Examine this code for multiple hard drives.
+    @hostname = parse_single_powershell_value(@winrm.run_powershell_script(host_script))
+  end
+
+  def vm_harddisks(vm_name)
+    location_script = <<-location_eol
+    Get-SCVirtualHardDisk -VMMServer localhost -VM "#{vm_name}" | \
+     Select-Object -ExpandProperty Location
+    location_eol
+
+    @vhd_type ||= vm_vhdtype(vm_name)
+    if @vhd_type == "DynamicallyExpanding"
+      @vhd = parse_single_powershell_value(@winrm.run_powershell_script(location_script))
+    elsif @vhd_type == "Differencing"
+      # TODO: return both the Parent Disk and the Checkpoint Disk
+      raise "Currently unsupported VHD Type #{@vhd_type}"
+    else
+      raise "Currently unsupported VHD Type #{@vhd_type}"
+    end
+  end
+
+  def vm_vhdtype(vm_name)
+    vhdtype_script = <<-vhdtype_eol
+    Get-SCVirtualHardDisk -VMMServer localhost -VM "#{vm_name}" | \
+      Select-Object -ExpandProperty VHDType
```

Comment:
@jerryk55 -
One thing to bear in mind…. Depending on the configuration of the VMM server, the powershell modules will reloaded into each session. This means you will need to explicitly include the line:
import-module virtualmachinemanager 
into your powershell scripts. I have not come across this myself but Nick Catling did (https://github.com/ManageIQ/manageiq/pull/2094)
You might want to test on his SCVMM server. LMK if you need the login details.